compiler/rustc_trait_selection/src/traits/error_reporting/type_err_ctxt_ext.rs

```diff
@@ -639,6 +639,32 @@ impl<'tcx> TypeErrCtxt<'_, 'tcx> {
                             err.span_label(span, explanation);
                         }
 
+                        if Some(trait_ref.def_id()) == self.tcx.lang_items().sized_trait() {
+                            if !self.tcx.object_safety_violations(trait_ref.def_id()).is_empty() {
```

Comment:
`Sized` is never object-safe. Why the need to check the `object_safe_violations`?

Reply:
Could you maybe add a comment outlining the motivation for delaying the error in these cases for future readers?

Reply:
This is now completely gone.